linux/shim/client_linux.go

```diff
@@ -2,9 +2,38 @@
 
 package shim
 
-import "syscall"
+import (
+	"context"
+	"os/exec"
+	"syscall"
+
+	"github.com/containerd/cgroups"
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
 
 var atter = syscall.SysProcAttr{
 	Cloneflags: syscall.CLONE_NEWNS,
 	Setpgid:    true,
 }
+
+func setCgroup(ctx context.Context, config Config, cmd *exec.Cmd) error {
+	cg, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(config.CgroupPath))
+	if err != nil {
+		terminate(cmd)
+		return errors.Wrapf(err, "failed to load cgroup %s", config.CgroupPath)
+	}
+	if err := cg.Add(cgroups.Process{
+		Pid: cmd.Process.Pid,
+	}); err != nil {
+		cg.Delete()
```

Comment:
Does `cg.Delete` delete the cgroup?
Since the cgroup is created by the client, I don't think we want to remove the cgroup inside containerd.